utils/create_dated_directory.py

```diff
@@ -0,0 +1,41 @@
+"""
+create_dated_dir Module
+
+This module provides a function for creating dated directories in the source and target directories
+based on the current date. The created directories follow a specific naming convention with the
+current date and an execution number.
+
+"""
+
+import datetime as dt
+import os
+
+
+def producer_dated_dir(source_dir: str, target_dir: str) -> tuple:
+    """
+    Create dated directories in the source and target directories based on the current date.
+
+    Parameters:
+    - source_dir (str): The source directory where the dated subdirectories will be created.
+    - target_dir (str): The target directory where corresponding dated subdirectories will be created.
+
+    Returns:
+    tuple: A tuple containing the paths of the newly created source and target directories.
+    """
+
+    today_date = dt.datetime.now().strftime("%d%m%Y")
+
+    try:
+        folders_date = list(filter(lambda x: today_date in x, os.listdir(source_dir)))
```

Comment:
i don't think so, this is correct logic for identifying exection number, as firstly it checks in both files and directory

> You should first filter out directory from os.listdir, then only do some operation.

Also, just doing brute_force check using `in` is not good, as date can occur in any format like foldername_01022024 which is not something we are looking for

> Instead use regex or startswith to verify correct format

And, finally it also possible to delete some folder in between some execution like 4 and 6, in such case total files will be 5, and execution number will be decided as 6, but dated folder for execution number 6 already exists, hence it is not good to use `len(folders_date)` to count last execution numbers

> Instead, filters current day files using regex as suggested above, and find highest execution number, this ensures execution number is always incremented.